### PR TITLE
Add inner loop tests

### PR DESCRIPTION
**Pull request by DrewScoggins**

### CODE REVIEW

File: src/tools/ScenarioMeasurement/Util/ProcessHelper.cs [hunk 1]
```diff
@@ -135,7 +135,8 @@ public ProcessHelper(Logger logger)
                     if (!exited)
                     {
                         process.Kill();
-                        return (Result.CloseFailed, pid);
+                        Thread.Sleep(3000);
+                        return (Result.Success, pid);
```
Comment (billwert) [inline src/tools/ScenarioMeasurement/Util/ProcessHelper.cs:139]: Why is this a success case now?

File: eng/performance/sdk_scenarios.proj [hunk 2]
```diff
@@ -103,5 +103,15 @@
       <Timeout>4:00</Timeout>
     </HelixWorkItem>
   </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
+    <HelixWorkItem Include="SDK Inner Loop Console">
```
Comment (billwert) [inline eng/performance/sdk_scenarios.proj:108]: nit: "SDK" sort of implies ownership in the SDK team. Is that appropriate here?

File: src/scenarios/emptyconsoletemplateinnerloop/test.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+'''
+from shared.runner import TestTraits, Runner
+
+EXENAME = 'emptyconsoletemplateinnerloop'
+
+if __name__ == "__main__":
+    traits = TestTraits(exename=EXENAME, 
+                        startupmetric='InnerLoop',
+                        guiapp='false',
+                        innerloopcommandargs='-c "from shutil import copyfile; copyfile(\'src/Program.cs\', \'app/Program.cs\')"'
```
Comment (billwert) [inline src/scenarios/emptyconsoletemplateinnerloop/test.py:12]: should this use pathsep?

File: src/scenarios/shared/util.py [hunk 4]
```diff
@@ -34,6 +34,10 @@ def builtexe(exename: str):
     'gets binary path'
     return os.path.join(const.BINDIR, '%s%s' % (exename, extension()))
 
+def appfolder(projname: str):
+    'gets path for callign dotnet run'
```
Comment (billwert) [inline src/scenarios/shared/util.py:38]:
```suggestion
    'gets path for calling dotnet run'
```

File: src/scenarios/shared/util.py [hunk 5]
```diff
@@ -34,6 +34,10 @@ def builtexe(exename: str):
...
+    return os.path.join(const.APPDIR, '%s.csproj' % projname)
```
Comment (billwert) [inline src/scenarios/shared/util.py:39]: should this include the project extension? We have some VB floating around etc.

File: src/tools/ScenarioMeasurement/Startup/Startup.cs [hunk 6]
```diff
@@ -161,13 +183,48 @@ static void checkArg(string arg, string name)
...
+
+                            logger.LogIterationHeader($"Iteration {i} - Diff");
+                            iterationResult = RunIteration(null, TestProcess, cleanupProcHelper, logger);
+                            if (!iterationResult.Success)
+                            {
+                                failed = true;
+                                break;
+                            }
+                            pids.Add(iterationResult.Pid);
+                            InnerLoopMarkerEventSource.Log.EndIteration();
```
Comment (billwert) [inline src/tools/ScenarioMeasurement/Startup/Startup.cs:225]: could this be just after `RunIteration`?

Comment (billwert) [reply src/tools/ScenarioMeasurement/Startup/Startup.cs:225]: oh, I guess it doesn't matter since these are just fences and not measurement spots.

**Review by billwert — CHANGES_REQUESTED** [review]
Some questions, but otherwise looks great!

File: src/tools/ScenarioMeasurement/Util/ProcessHelper.cs (hunk 1 above)
Comment (DrewScoggins) [reply src/tools/ScenarioMeasurement/Util/ProcessHelper.cs:139]: This is dead code that I need to rip out. Good catch!

File: src/tools/ScenarioMeasurement/Startup/Startup.cs (hunk 6 above)
Comment (DrewScoggins) [reply src/tools/ScenarioMeasurement/Startup/Startup.cs:225]: Yeah, I just use them as fences to let me know which side of the test we are in.

File: eng/performance/sdk_scenarios.proj (hunk 2 above)
Comment (DrewScoggins) [reply eng/performance/sdk_scenarios.proj:108]: Yeah, this was just a copy paste. I will change this.

File: src/scenarios/emptyconsoletemplateinnerloop/test.py (hunk 3 above)
Comment (DrewScoggins) [reply src/scenarios/emptyconsoletemplateinnerloop/test.py:12]: Yeah, I should probably use os.join to create these paths.

File: src/scenarios/shared/util.py (hunk 5 above)
Comment (DrewScoggins) [reply src/scenarios/shared/util.py:39]: Done, had to solve this by adding a new test trait, but it is pretty clean.

**Review by billwert — APPROVED** [review]
LGTM!

**Pull request merged**